Tests/FluentAssertions.Specs/Xml/XElementAssertionSpecs.cs

```diff
@@ -1228,5 +1228,185 @@ public void When_asserting_element_has_a_child_element_with_an_empty_name_it_sho
         }
 
         #endregion
+
+        #region BeSingle
+
+        [Fact]
+        public void When_asserting_document_has_a_single_child_element_and_it_does_it_should_succeed()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                  </parent>");
+
+            // Act / Assert
+            document.Should().HaveElement("child").Which.Should().BeSingle();
+        }
+
+        [Fact]
+        public void When_asserting_document_has_single_child_element_but_it_does_have_two_it_should_fail()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                  </parent>");
+
+            // Act
+            Action act = () => document.Should().HaveElement("child").Which.Should().BeSingle();
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage(
+                "Expected child to have a count of 1, but found 2.");
```

Comment:
I'm missing "document", the "sut", in the message.
All assertions from FluentAssertions should have the subject (everything before `.Should()`) in the message.
My proposal: `Expected document to have a single element "child", but found 2.`

Reply:
The normal "Should().HaveCount()" assertion uses "item(s)", so I added the subject and use "1 element(s)" instead of "single element".

Reply:
Maybe all assertions on `XDocument` does not follow the convention to include the "sut" in the message.
@dennisdoomen ?